From 94b389a3301ec5b644b0b396e32d29fd27aa0a91 Mon Sep 17 00:00:00 2001
From: "Iain R. Learmonth" <irl@fsfe.org>
Date: Wed, 30 Aug 2017 23:12:24 +0100
Subject: [PATCH] Fix test for funcfiletrace on later zshdb in ok4zshdb.sh
 (Closes: #9)

funcfiletrace is no longer listed in the output of declare -p.
---
 test/zsh/ok4zshdb.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/zsh/ok4zshdb.sh b/test/zsh/ok4zshdb.sh
index 97c777d..bf87f6f 100755
--- a/test/zsh/ok4zshdb.sh
+++ b/test/zsh/ok4zshdb.sh
@@ -4,8 +4,8 @@ PS4='%(%x:%I): [%?]
 '
 
 second_fn() {
-  decls=$(declare -p)
-  if [[ $decls != *funcfiletrace* ]] ; then
+  zmodload zsh/parameter
+  if ! (( ${+funcfiletrace} )) ; then
     print "Looks like you don't have funcfiletrace."
     print "We need a zsh new enough which has that."
     exit 10
